tests/kernel/timer/timer_monotonic/src/main.c

```diff
@@ -77,11 +195,153 @@ void test_timer(void)
 
 	zassert_false(errors, "errors = %d\n", errors);
 
-	zassert_false(test_frequency(), "test frequency failed");
+	zassert_false(test_frequency(1000,100U),
+			"test frequency failed");
+}
+
+/**
+ * @brief Test monotonic timer
+ *
+ * Validates monotonic timer's clock calibration for 1 ms.
+ *
+ * It reads the System clock’s h/w timer frequency value continuously
+ * using k_cycle_get_32() to verify its working and correctiveness.
+ * It also checks system tick frequency by checking the delta error
+ * between generated and system clock provided HW cycles per msec values.
+ *
+ * @ingroup kernel_timer_tests
+ *
+ * @see k_cycle_get_32(), sys_clock_hw_cycles_per_sec()
+ */
+void test_timer_msec(void)
+{
```

Comment:
I don't understand this test: it looks like a simple copy of the existing `test_timer()` case, except for the argument to test_frequency() at the end.  What's different?

Why?  Why not just put a second call to test_frequency() into the original case, or split test_frequency() out into a separate test case (or cases) instead of putting in another million-iteration (!!!) loop into the code?

Reply:
Thanks for your comments. and I have updated the code to split the function:”test_frequency()” to 2 functions to support 2 different test cases. Could you please help to review it again ? Thanks again.

Reply:
I still don't understand what behavior this is testing.  It looks like it's doing exactly the same thing as `test_timer()` already did.  And it's very expensive, being a million-iteration spinning loop.  Why do we want this?